v2/controllers/egress_watcher.go

```diff
@@ -55,8 +55,27 @@ func (r *EgressWatcher) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		return ctrl.Result{}, err
 	}
 
+	targetPods := make(map[string]*corev1.Pod)
 	for _, pod := range pods.Items {
-		for k, v := range pod.Annotations {
+		if pod.Spec.HostNetwork {
+			// Pods in host network cannot use egress NAT.
+			// So skip it.
+			continue
+		}
+		podIp := pod.Status.PodIP
+		// The reconciliation should be triggered only for running pods.
+		if pod.Status.Phase == corev1.PodRunning {
+			if _, found := targetPods[podIp]; found {
+				// multiple running pods have the same address.
+				return ctrl.Result{}, fmt.Errorf("multiple pods have the same address: %s", podIp)
+			}
+			tmp := pod
```

Comment:
This creates a shallow copy.
It might be okay for this context, but I think the following is more preferable.
```go
for i := range pods.Items {
  pod := &pods.Items[i]
  ...
}
```

Reply:
Thank you!
I fixed it.